subsys/net/ip/ipv6_fragment.c

```diff
@@ -602,6 +602,8 @@ static int send_ipv6_fragment(struct net_pkt *pkt,
 		goto fail;
 	}
 
+	net_pkt_set_ipv6_next_hdr(frag_pkt, NET_IPV6_NEXTHDR_FRAG);
```

Comment:
Hum, that's not going to work if original header comes with optional headers.

Reply:
That's true, I will investigate further.

Reply:
Instead, have a 
`u8_t frag_pkt_next_hdr = NET_IPV6_NEXTHDR_HBHO;`
 declared 

instead of your line 605, have this:
```
if (!net_pkt_ipv6_ext_len(pkt)) {
    frag_pkt_next_hdr = NET_IPV6_NEXTHDR_FRAG;
}
```

and in line 637:
`if (net_ipv6_finalize(frag_pkt, frag_pkt_next_hdr) < 0) {`


That will fix the issue I think. Try on tests/net/ipv6_fragment and with an actual sample + ping

Reply:
It's working. Tried ping and tests. Can you send a patch, I will drop this PR.

Reply:
Ok